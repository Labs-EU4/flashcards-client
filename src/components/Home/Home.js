import React from "react";
import styles from "./Home.module.css";
import RecentDecks from "../RecentDecks/RecentDecks";
import Dashboard from "../../layout/Dashboard/Dashboard";
import CreateNewDeck from "../../components/NewDeckForm/NewDeckForm";

const Home = () => {
  return (
    <>
      <Dashboard>
        <div className={styles.leftContent}></div>
        <div className={styles.rightContent}>
          <RecentDecks />
          <CreateNewDeck />
        </div>
      </Dashboard>
    </>
  );
};

export default Home;
