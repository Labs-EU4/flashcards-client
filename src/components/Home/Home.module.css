.leftContent {
  width: 100%;
  background-color: #f9f9f9;
  margin-right: 2%;
  border-radius: 5px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  overflow-y: auto;
}

.rightContent {
  width: 100%;
  display: flex;
  flex-direction: column;
  margin-left: 2%;
  max-height: 100%;
}

@media only screen and (max-width: 1024px) {
  .leftContent {
    width: 96%;
    margin: 2%;
  }

  .rightContent {
    display: flex;
    flex-direction: column;
    width: 96%;
    margin: 2%;
  }
}
