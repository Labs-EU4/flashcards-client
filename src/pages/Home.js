import React from "react";
import BasicLayout from "../layout/BasicLayout";

const Home = () => {
  return <BasicLayout />;
};

export default Home;
