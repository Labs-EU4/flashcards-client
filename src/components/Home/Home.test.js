import React from "react";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Home from "./Home";
import {BrowserRouter} from "react-router-dom";

let wrapper;
beforeEach(() => {
  wrapper = rtl.render(
    <BrowserRouter>
      <Home />
    </BrowserRouter>
  );
});

afterEach(rtl.cleanup);

it("should take a snapshot", () => {
  expect(wrapper.container).toMatchSnapshot();
});

describe("is the component rendering correctly", () => {
  it("renders the left content", () => {
    let queryValue = wrapper.getByTestId("content-left");
    expect(queryValue).toBeInTheDocument();
  });

  it("renders the right content", () => {
    let queryValue = wrapper.getByTestId("content-right");
    expect(queryValue).toBeInTheDocument();
  });
});
