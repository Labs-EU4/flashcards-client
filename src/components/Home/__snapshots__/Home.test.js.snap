// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div>
    <section
      class="dashboard ant-layout ant-layout-has-sider"
    >
      <div
        class="dashboardLeft"
      >
        <aside
          class="sider ant-layout-sider ant-layout-sider-dark"
          style="flex: 0 0 200px; max-width: 200px; min-width: 200px; width: 200px;"
        >
          <div
            class="ant-layout-sider-children"
          >
            <div
              class="logoAndCollapse"
            >
              <a
                href="/"
              >
                <img
                  alt="logo"
                  class="logo"
                  data-testid="logo"
                  src="https://i.imgur.com/tuS7kwh.png"
                />
              </a>
              <i
                aria-label="icon: menu-fold"
                class="anticon anticon-menu-fold trigger"
                data-testid="toggle-icon"
                style="font-size: 24px;"
                tabindex="-1"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="menu-fold"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"
                  />
                </svg>
              </i>
            </div>
            <h3
              class="greeting"
              data-testid="greeting"
            >
              Welcome, 
              !
            </h3>
            <ul
              class="ant-menu menu ant-menu-light ant-menu-root ant-menu-inline"
              data-testid="menu"
              role="menu"
            >
              <li
                class="ant-menu-item menuItem"
                role="menuitem"
                style="padding-left: 24px;"
              >
                <a
                  aria-current="page"
                  class="navItem navItemSeleted"
                  href="/"
                >
                  <i
                    aria-label="icon: home"
                    class="anticon anticon-home"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="home"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                      />
                    </svg>
                  </i>
                  <span>
                    Home
                  </span>
                </a>
              </li>
              <li
                class="ant-menu-item menuItem"
                role="menuitem"
                style="padding-left: 24px;"
              >
                <a
                  class="navItem"
                  href="/deck-library"
                >
                  <i
                    aria-label="icon: block"
                    class="anticon anticon-block"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="block"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M856 376H648V168c0-8.8-7.2-16-16-16H168c-8.8 0-16 7.2-16 16v464c0 8.8 7.2 16 16 16h208v208c0 8.8 7.2 16 16 16h464c8.8 0 16-7.2 16-16V392c0-8.8-7.2-16-16-16zm-480 16v188H220V220h360v156H392c-8.8 0-16 7.2-16 16zm204 52v136H444V444h136zm224 360H444V648h188c8.8 0 16-7.2 16-16V444h156v360z"
                      />
                    </svg>
                  </i>
                  <span>
                    Deck Library
                  </span>
                </a>
              </li>
              <li
                class="ant-menu-item menuItem"
                role="menuitem"
                style="padding-left: 24px;"
              >
                <a
                  class="navItem"
                  href="/discover-decks"
                >
                  <i
                    aria-label="icon: global"
                    class="anticon anticon-global"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="global"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M854.4 800.9c.2-.3.5-.6.7-.9C920.6 722.1 960 621.7 960 512s-39.4-210.1-104.8-288c-.2-.3-.5-.5-.7-.8-1.1-1.3-2.1-2.5-3.2-3.7-.4-.5-.8-.9-1.2-1.4l-4.1-4.7-.1-.1c-1.5-1.7-3.1-3.4-4.6-5.1l-.1-.1c-3.2-3.4-6.4-6.8-9.7-10.1l-.1-.1-4.8-4.8-.3-.3c-1.5-1.5-3-2.9-4.5-4.3-.5-.5-1-1-1.6-1.5-1-1-2-1.9-3-2.8-.3-.3-.7-.6-1-1C736.4 109.2 629.5 64 512 64s-224.4 45.2-304.3 119.2c-.3.3-.7.6-1 1-1 .9-2 1.9-3 2.9-.5.5-1 1-1.6 1.5-1.5 1.4-3 2.9-4.5 4.3l-.3.3-4.8 4.8-.1.1c-3.3 3.3-6.5 6.7-9.7 10.1l-.1.1c-1.6 1.7-3.1 3.4-4.6 5.1l-.1.1c-1.4 1.5-2.8 3.1-4.1 4.7-.4.5-.8.9-1.2 1.4-1.1 1.2-2.1 2.5-3.2 3.7-.2.3-.5.5-.7.8C103.4 301.9 64 402.3 64 512s39.4 210.1 104.8 288c.2.3.5.6.7.9l3.1 3.7c.4.5.8.9 1.2 1.4l4.1 4.7c0 .1.1.1.1.2 1.5 1.7 3 3.4 4.6 5l.1.1c3.2 3.4 6.4 6.8 9.6 10.1l.1.1c1.6 1.6 3.1 3.2 4.7 4.7l.3.3c3.3 3.3 6.7 6.5 10.1 9.6 80.1 74 187 119.2 304.5 119.2s224.4-45.2 304.3-119.2a300 300 0 0 0 10-9.6l.3-.3c1.6-1.6 3.2-3.1 4.7-4.7l.1-.1c3.3-3.3 6.5-6.7 9.6-10.1l.1-.1c1.5-1.7 3.1-3.3 4.6-5 0-.1.1-.1.1-.2 1.4-1.5 2.8-3.1 4.1-4.7.4-.5.8-.9 1.2-1.4a99 99 0 0 0 3.3-3.7zm4.1-142.6c-13.8 32.6-32 62.8-54.2 90.2a444.07 444.07 0 0 0-81.5-55.9c11.6-46.9 18.8-98.4 20.7-152.6H887c-3 40.9-12.6 80.6-28.5 118.3zM887 484H743.5c-1.9-54.2-9.1-105.7-20.7-152.6 29.3-15.6 56.6-34.4 81.5-55.9A373.86 373.86 0 0 1 887 484zM658.3 165.5c39.7 16.8 75.8 40 107.6 69.2a394.72 394.72 0 0 1-59.4 41.8c-15.7-45-35.8-84.1-59.2-115.4 3.7 1.4 7.4 2.9 11 4.4zm-90.6 700.6c-9.2 7.2-18.4 12.7-27.7 16.4V697a389.1 389.1 0 0 1 115.7 26.2c-8.3 24.6-17.9 47.3-29 67.8-17.4 32.4-37.8 58.3-59 75.1zm59-633.1c11 20.6 20.7 43.3 29 67.8A389.1 389.1 0 0 1 540 327V141.6c9.2 3.7 18.5 9.1 27.7 16.4 21.2 16.7 41.6 42.6 59 75zM540 640.9V540h147.5c-1.6 44.2-7.1 87.1-16.3 127.8l-.3 1.2A445.02 445.02 0 0 0 540 640.9zm0-156.9V383.1c45.8-2.8 89.8-12.5 130.9-28.1l.3 1.2c9.2 40.7 14.7 83.5 16.3 127.8H540zm-56 56v100.9c-45.8 2.8-89.8 12.5-130.9 28.1l-.3-1.2c-9.2-40.7-14.7-83.5-16.3-127.8H484zm-147.5-56c1.6-44.2 7.1-87.1 16.3-127.8l.3-1.2c41.1 15.6 85 25.3 130.9 28.1V484H336.5zM484 697v185.4c-9.2-3.7-18.5-9.1-27.7-16.4-21.2-16.7-41.7-42.7-59.1-75.1-11-20.6-20.7-43.3-29-67.8 37.2-14.6 75.9-23.3 115.8-26.1zm0-370a389.1 389.1 0 0 1-115.7-26.2c8.3-24.6 17.9-47.3 29-67.8 17.4-32.4 37.8-58.4 59.1-75.1 9.2-7.2 18.4-12.7 27.7-16.4V327zM365.7 165.5c3.7-1.5 7.3-3 11-4.4-23.4 31.3-43.5 70.4-59.2 115.4-21-12-40.9-26-59.4-41.8 31.8-29.2 67.9-52.4 107.6-69.2zM165.5 365.7c13.8-32.6 32-62.8 54.2-90.2 24.9 21.5 52.2 40.3 81.5 55.9-11.6 46.9-18.8 98.4-20.7 152.6H137c3-40.9 12.6-80.6 28.5-118.3zM137 540h143.5c1.9 54.2 9.1 105.7 20.7 152.6a444.07 444.07 0 0 0-81.5 55.9A373.86 373.86 0 0 1 137 540zm228.7 318.5c-39.7-16.8-75.8-40-107.6-69.2 18.5-15.8 38.4-29.7 59.4-41.8 15.7 45 35.8 84.1 59.2 115.4-3.7-1.4-7.4-2.9-11-4.4zm292.6 0c-3.7 1.5-7.3 3-11 4.4 23.4-31.3 43.5-70.4 59.2-115.4 21 12 40.9 26 59.4 41.8a373.81 373.81 0 0 1-107.6 69.2z"
                      />
                    </svg>
                  </i>
                  <span>
                    Discover Decks
                  </span>
                </a>
              </li>
            </ul>
            <footer
              class="footer"
              data-testid="footer"
            >
              <a
                href="/login"
              >
                <button
                  class="ant-btn logoutButton ant-btn-primary"
                  data-testid="logout-button"
                  type="button"
                >
                  <span>
                    Logout
                  </span>
                </button>
              </a>
              <p>
                <i
                  aria-label="icon: setting"
                  class="anticon anticon-setting"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="setting"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 0 0 9.3-35.2l-.9-2.6a443.74 443.74 0 0 0-79.7-137.9l-1.8-2.1a32.12 32.12 0 0 0-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 0 0-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 0 0-25.8 25.7l-15.8 85.4a351.86 351.86 0 0 0-99 57.4l-81.9-29.1a32 32 0 0 0-35.1 9.5l-1.8 2.1a446.02 446.02 0 0 0-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 0 0-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0 0 35.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0 0 25.8 25.7l2.7.5a449.4 449.4 0 0 0 159 0l2.7-.5a32.05 32.05 0 0 0 25.8-25.7l15.7-85a350 350 0 0 0 99.7-57.6l81.3 28.9a32 32 0 0 0 35.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 0 1-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 0 1-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 0 1 512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 0 1 400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 0 1 624 502c0 29.9-11.7 58-32.8 79.2z"
                    />
                  </svg>
                </i>
                 Profile Settings
              </p>
            </footer>
          </div>
        </aside>
      </div>
      <section
        class="ant-layout"
      >
        <main
          class="content ant-layout-content"
        >
          <div
            class="leftContent"
            data-testid="content-left"
          >
            <div
              class="DeckCardContainer"
            >
              <h1
                class="heading"
              >
                My Decks
              </h1>
              <div
                class="noDecks"
              >
                You have no decks
              </div>
            </div>
          </div>
          <div
            class="rightContent"
            data-testid="content-right"
          >
            <div
              class="container"
            >
              <div
                class="ant-spin-nested-loading"
              >
                <div>
                  <div
                    class="ant-spin ant-spin-spinning"
                  >
                    <span
                      class="ant-spin-dot ant-spin-dot-spin"
                    >
                      <i
                        class="ant-spin-dot-item"
                      />
                      <i
                        class="ant-spin-dot-item"
                      />
                      <i
                        class="ant-spin-dot-item"
                      />
                      <i
                        class="ant-spin-dot-item"
                      />
                    </span>
                  </div>
                </div>
                <div
                  class="ant-spin-container ant-spin-blur"
                >
                  <h1
                    class="heading"
                    data-testid="heading"
                  >
                    Recent decks
                  </h1>
                  <div
                    data-testid="decks"
                  >
                    <div
                      class="noDecks"
                    >
                      You haven't completed a session yet!
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="containerCreateDeck"
            >
              <h1
                class="heading"
              >
                Create New Deck
              </h1>
              <form
                class="ant-form ant-form-horizontal login-form"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control-wrapper"
                  >
                    <div
                      class="ant-form-item-control"
                    >
                      <span
                        class="ant-form-item-children"
                      >
                        <input
                          class="ant-input"
                          data-__field="[object Object]"
                          data-__meta="[object Object]"
                          data-testid="inputDeckName"
                          id="newDeckForm_text"
                          name="name"
                          placeholder="Deckname"
                          type="text"
                          value=""
                        />
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control-wrapper"
                  >
                    <div
                      class="ant-form-item-control"
                    >
                      <span
                        class="ant-form-item-children"
                      >
                        <div
                          class="ant-select ant-select-enabled ant-select-allow-clear"
                          style="width: 100%;"
                        >
                          <div
                            aria-autocomplete="list"
                            aria-controls="test-uuid"
                            aria-expanded="false"
                            aria-haspopup="true"
                            class="ant-select-selection
            ant-select-selection--multiple"
                            data-testid="inputSelect"
                            role="combobox"
                          >
                            <div
                              class="ant-select-selection__rendered"
                            >
                              <ul>
                                <li
                                  class="ant-select-search ant-select-search--inline"
                                >
                                  <div
                                    class="ant-select-search__field__wrap"
                                  >
                                    <input
                                      autocomplete="off"
                                      class="ant-select-search__field"
                                      value=""
                                    />
                                    <span
                                      class="ant-select-search__field__mirror"
                                    >
                                      
                                       
                                    </span>
                                  </div>
                                </li>
                              </ul>
                            </div>
                          </div>
                        </div>
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control-wrapper"
                  >
                    <div
                      class="ant-form-item-control"
                    >
                      <span
                        class="ant-form-item-children"
                      >
                        <label
                          class="ant-checkbox-wrapper"
                        >
                          <span
                            class="ant-checkbox"
                          >
                            <input
                              class="ant-checkbox-input"
                              data-testid="inputCheck"
                              type="checkbox"
                              value=""
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                          <span>
                            Public
                          </span>
                        </label>
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control-wrapper"
                  >
                    <div
                      class="ant-form-item-control"
                    >
                      <span
                        class="ant-form-item-children"
                      >
                        <button
                          class="ant-btn ant-btn-primary"
                          data-testid="button"
                          id="create-deck-button"
                          type="submit"
                        >
                          <span>
                            Create Deck
                          </span>
                        </button>
                      </span>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </main>
      </section>
    </section>
  </div>
</div>
`;
